### PR TITLE
ci(flake8): enable unused import check

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Changes
This PR enables a `flake8` test to verify unused import and fixes all the occurrences we currently have.

## How did you test this code?
I didn't. I hope unit/integration tests will catch any regression

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -81,3 +80,12 @@ per-file-ignores =
     ./posthog/settings/overrides.py: T001
     ./posthog/utils.py: T001
     ./posthog/async_migrations/test/test_migrations_not_required.py: T001
+
+    ./ee/clickhouse/materialized_columns/__init__.py: F401
```
Comment (hazzadous) [inline .flake8:84]: Could we add these ignores to the actual file? This way we make it very obvious to a developer working on these files what is applicable or not. It also means we don't need to keep this list in sync with e.g. files moving around or deleted

**Comment by Twixes** [comment]
I'm actually hopeful we can get this in!

**Comment by guidoiaquinti** [comment]
> I'm actually hopeful we can get this in!

Ahaha yeah I'm hopeful too despite it is just a personal side quest and I don't think I'll have time to investigate those failures this sprint! Feel free to push directly in this branch if you want to contribute!

**Comment by macobo** [comment]
Q: Does flake8 pre-commit hook autofix these errors?

**Comment by guidoiaquinti** [comment]
> Q: Does flake8 pre-commit hook autofix these errors?

It doesn't fix but it reports them.

**Review by hazzadous — APPROVED** [review]
Looks good

re `flake8 -fix` previously have done some convoluted isort / flake8 / autoflake dance to auto remove unused imports. We could run that on pre-commit, but as a first port of call I'd just make sure that pre-commit warns appropriately so we don't have to wait for CI

**Pull request merged**